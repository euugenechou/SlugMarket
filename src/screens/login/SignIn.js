/* React imports */
import React from "react";
import {
  TextInput,
  StyleSheet,
  View,
  Text,
  ScrollView
} from "react-native";
import TopAlert from "../components/TopAlert";

/* AWS imports */
import { Auth } from "aws-amplify";
import Expo, { Font } from 'expo';
import { Button } from 'react-native-elements'

export default class App extends React.Component {
  static navigationOptions = ({ navigation }) => ({
    header: null
  });

  constructor(props) {
    super(props);
    this.state = {
      username: "",
      password: "",
      confirmationCode: "",
      user: {},
      visible: false,
      errorMessage: "",
      isAuthenticated: false
    };
  }

  onChangeText(key, value) {
    this.setState({
      [key]: value
    });
  }

  signIn() {
    const { username, password } = this.state;
    Auth.signIn(username, password)
      .then(() => {
        user => this.setState({ user });
        this.props.navigation.push("MainScreen");
      })
      .catch(() =>
        this.setState({
          visible: true,
          errorMessage: "Username or password incorrect"
        })
      );
  }

  render() {
    return (
      <View style={{flex: 1}}>
        <TopAlert
          visible={this.state.visible}
          message={this.state.errorMessage}
        />
        <ScrollView
          contentContainerStyle={styles.container}
          centerContent={true}
        >
          <Text style={styles.titleText}>SlugMarket</Text>
          <TextInput
            onChangeText={value => this.onChangeText("username", value)}
            style={styles.input}
            placeholder="UCSC email"
            placeholderTextColor="gray"
            autoCapitalize="none"
            autoCorrect={false}
          />
          <TextInput
            onChangeText={value => this.onChangeText("password", value)}
            style={styles.input}
            secureTextEntry={true}
            placeholder="password"
            placeholderTextColor="gray"
            autoCapitalize="none"
            autoCorrect={false}
          />
          <Button
            raised
            color='white'
            title="Sign In"
            fontWeight="bold"
            onPress={() => this.signIn()}
            backgroundColor="teal"
            borderRadius={5}
            containerViewStyle={{width: 300, paddingVertical: 20}}
            />
          <Button
            raised
            color="white"
            title="Sign Up"
            fontWeight="bold"
            backgroundColor="teal"
            borderRadius={5}
            containerViewStyle={{width: 300}}
            onPress={() => {
              this.setState({ visible: false });
              this.props.navigation.navigate("SignUpScreen");
            }}
          />
        </ScrollView>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  input: {
    justifyContent: "center",
    alignSelf: "center",
    height: 50,
    color: "black",
    borderBottomWidth: 0.7,
    borderBottomColor: "darkgray",
    marginBottom: 20,
    width: 300
  },
  container: {
    flex: 1,
    paddingVertical: 150,
    backgroundColor: "white",
    alignItems: "center",
  },
  titleText: {
    fontSize: 36,
    fontWeight: "700",
    padding: 30,
    color: "black",
    textShadowColor: "#dddddd",
    textShadowRadius: 15,
  }
});
