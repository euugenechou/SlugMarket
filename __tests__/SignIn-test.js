// __tests__/SignIn-test.js
import "react-native";
import React from "react";
import SignIn from "../src/screens/login/SignIn";

import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<SignIn />).toJSON();
  expect(tree).toMatchSnapshot();
});
