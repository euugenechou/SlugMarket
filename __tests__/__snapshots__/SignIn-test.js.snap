// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    centerContent={true}
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flex": 1,
        "paddingVertical": 150,
      }
    }
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "fontSize": 36,
            "fontWeight": "700",
            "padding": 30,
            "textShadowColor": "#dddddd",
            "textShadowRadius": 15,
          }
        }
      >
         SlugMarket 
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        onChangeText={[Function]}
        placeholder="UCSC email"
        placeholderTextColor="gray"
        style={
          Object {
            "alignSelf": "center",
            "borderBottomColor": "darkgray",
            "borderBottomWidth": 0.7,
            "color": "black",
            "height": 50,
            "justifyContent": "center",
            "marginBottom": 20,
            "width": 300,
          }
        }
      />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        onChangeText={[Function]}
        placeholder="password"
        placeholderTextColor="gray"
        secureTextEntry={true}
        style={
          Object {
            "alignSelf": "center",
            "borderBottomColor": "darkgray",
            "borderBottomWidth": 0.7,
            "color": "black",
            "height": 50,
            "justifyContent": "center",
            "marginBottom": 20,
            "width": 300,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "marginLeft": 15,
              "marginRight": 15,
            },
            Object {
              "shadowColor": "rgba(0,0,0, .4)",
              "shadowOffset": Object {
                "height": 1,
                "width": 1,
              },
              "shadowOpacity": 1,
              "shadowRadius": 1,
            },
            Object {
              "paddingVertical": 20,
              "width": 300,
            },
            Object {
              "borderRadius": 5,
            },
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            pointerEvents="box-only"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "teal",
                },
                Object {
                  "borderRadius": 5,
                },
                Object {
                  "padding": 12,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 20,
                    },
                    Object {
                      "color": "white",
                    },
                    Object {
                      "fontSize": 17.5,
                    },
                    undefined,
                    undefined,
                    Object {
                      "fontWeight": "bold",
                    },
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Sign In
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "marginLeft": 15,
              "marginRight": 15,
            },
            Object {
              "shadowColor": "rgba(0,0,0, .4)",
              "shadowOffset": Object {
                "height": 1,
                "width": 1,
              },
              "shadowOpacity": 1,
              "shadowRadius": 1,
            },
            Object {
              "width": 300,
            },
            Object {
              "borderRadius": 5,
            },
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            pointerEvents="box-only"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "teal",
                },
                Object {
                  "borderRadius": 5,
                },
                Object {
                  "padding": 12,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 20,
                    },
                    Object {
                      "color": "white",
                    },
                    Object {
                      "fontSize": 17.5,
                    },
                    undefined,
                    undefined,
                    Object {
                      "fontWeight": "bold",
                    },
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Sign Up
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
